paper/automated.tex

```diff
@@ -1,20 +1,108 @@
+Tactics in Lean, on the other hand, are meta-programs~\cite{DBLP:journals/pacmpl/EbnerURAM17} that builds proofs.
+In other words, tactics are programs that operate at the meta-level of Lean code: they essentially take in Lean code as input and produce Lean code as output.
+In this manner, they look like another keyword in the language, and are tightly integrated by producing proofs directly.
+Under the hood, they can implement essentially anything, from syntactic sugar to full decision procedures.
+The \texttt{duper} tactic~\cite{DBLP:conf/itp/CluneQBA24}, for example, implements a superposition calculus, similar Vampire's, but for dependent types --- Lean's underlying logical foundation.
 
-\begin{itemize}
-    \item Z3 prover
-    \item Vampire
-    \item More elementary substitutions
-    \item egg
-    \item \href{https://leanprover.zulipchat.com/#narrow/channel/458659-Equational/topic/Austin.20pairs/near/479643838}{duper}
-\end{itemize}
+In the rest of this section we describe the different proof automation techniques and ATPs/tactics we used.
+We first discuss the different proof methods used: primarily superposition and equational reasoning, we then discuss the integration in Lean, and finally we report some basic empirical results from this project.
 
-Any comparative study of semi-automated methods with fully automated ones? In principle, the semi-automated approach could be more automated using a script or "agent" to call various theorem provers. See \href{https://leanprover.zulipchat.com/#narrow/stream/458659-Equational/topic/A.20magma.20of.20order.20.3C.2013.20-.20for.20Equation2531.3F}{this discussion}.
+\subsection{Proof Techniques}
 
-Draw upon the discussion \href{https://leanprover.zulipchat.com/#narrow/stream/458659-Equational/topic/Future.20of.20Using.20ATPs}{here} on the various ways of integrating ATPs with Lean. This project primarily used the least integrated approach, "Option 3", as it was fastest and required no dependencies on the other contributors, but it also has drawbacks.
+The main two families of ATPs and tactics we used here are superposition/saturation-based and equational reasoning ones.
+In this context we also include SMT solvers, which combine specific decision procedures for theories, like congruence closure for equational reasoning, with satisfiability (SAT) solving~\cite{}.
+Finally, we also used \texttt{aesop}~\cite{DBLP:conf/cpp/LimpergF23}, which implements a version of tableau search.
+This was used mainly to help specific constructions in refutations, and is not specific to proving or disproving magma implications in this sense.
+We will describe our use of \texttt{aesop} more in Section~\ref{sec:proof-reconstruction} below.
+
+\paragraph{Saturation}
+Most of the ATPs we used extensively in this project rely primarily on saturation procdures in the superposition calculus.
+This is the case for Vampire~\cite{DBLP:conf/cav/KovacsV13}, and the paper alo gives a more in-depth explanation of this.
```

Comment:
alo -> also. though i'd probably rephrase this to use the citation as the object in that clause.